gstd/gstd.c

```diff
@@ -200,12 +123,16 @@ main (gint argc, gchar * argv[])
   g_option_context_add_main_entries (context, entries, NULL);
 
   /* Initialize GStreamer */
-  gstreamer_group = gst_init_get_option_group ();
+  gstd_manager_new (supported_ipcs, num_ipcs, &manager,
+      &gstreamer_group, 0, NULL);
   g_option_context_add_group (context, gstreamer_group);
 
   /* Read option group for each IPC */
-  ipc_add_option_groups (ipc_array, supported_ipcs, num_ipcs, context,
-      optiongroup_array);
+  ipc_group_array = g_malloc0 (num_ipcs * sizeof (*ipc_group_array));
+  gstd_manager_ipc_options (manager, ipc_group_array);  /* If you don't want this option, you can avoid calling this function */
+  for (int i = 0; i < num_ipcs; i++) {
+    g_option_context_add_group (context, ipc_group_array[i]);
+  }
```

Comment:
We are leaking `ipc_group_array`, `g_option_context_add_group` only takes ownership of the members and not the memory from `g_malloc0` in line 131